### DISCUSSION TITLE
c# Supabase.Client issue with retrieving CurrentUser

### DISCUSSION THREAD

**Discussion by stefanE87** [supabase/supabase#27989] [Questions]
Hi all,

I am using Supabase C# client in a Blazor server (interactive) app on .NET 8 and use it for Authentication and the Postgres db I usewith EntityFramework.

Currently, I perform _supabaseClient.Auth.SignIn(email, password) and use session.User to build my ClaimsPrincipal of which I keep a reference in a custom AuthenticationStateProvider.

It works like a charm, but since blazor server works with SignalR, every time there is a page refresh a user needs to log in again since the circuit was destroyed (and also the instance of the custom AuthenticationStateProvider) . I tried using a cookie, but that seems to be a nightmare when not using the scaffolding for Identities platform.

Checking what I could try next I noticed the Supabase.Client.Auth.CurrentUser . When debugging could retrieve my user on a page refresh and set it again. The issue I have is that since the client is registered as a singleton, so I believe that if multiple users use the app it would always return the user of the first person who signed in. the supabase client needs to be registered as a singleton so I assume this is the reason, I have no idea how the state is managed of that class when multiple people start using the app.

I feel like I am not using the supabase client correctly/to it's full potential, is there anything I missed to avoid losing authentication on a page refresh?

**Comment by hansaliyad1** [comment]
Did you manage to figure this out?
